qgepqwat2ili/gui/__init__.py

```diff
@@ -198,12 +200,47 @@ def action_do_export():
             return
         progress_dialog.setValue(25)
 
+        tempdir = tempfile.TemporaryDirectory()
+        labels_file_path = os.path.join(tempdir.name, "labels.geojson")
+
+        # Export the labels file
+        progress_dialog.setLabelText("Extracting labels...")
+
+        structures_lyrs = QgsProject.instance().mapLayersByName("vw_qgep_wastewater_structure")
+        reaches_lyrs = QgsProject.instance().mapLayersByName("vw_qgep_reach")
+        if len(structures_lyrs) == 0 or len(reaches_lyrs) == 0:
+            progress_dialog.close()
+            show_failure(
+                "Could not find the vw_qgep_wastewater_structure and/or the vw_qgep_reach layers.",
+                "Make sure your QGEP project is open.",
+                None,
+            )
+            return
+        structures_lyr = structures_lyrs[0]
+        reaches_lyr = reaches_lyrs[0]
+
+        QApplication.processEvents()
```

Comment:
why do you need this?

Reply:
Slightly improves UX (otherwise not properly redrawing when the progress dialog changes size). Quite minor though, but since it was already done like this for other updates, I think it's fine to leave it.